minor stuff obstructing editions in godot 4.x

tehKaiN opened the pull request:
Related to #20 

Doesn't fix anything in waterway rendering, just the edit controls

tehKaiN commented:
Oops, I see that my godot editor trimmed all trailing whitespace. Is that ok for you, or should I reduce the diff?

TokisanGames commented on addons/waterways/gui/river_gizmo.gd:
```diff
@@ -253,24 +253,24 @@ func _set_handle(gizmo: EditorNode3DGizmo, index: int, secondary: bool, camera:
-			var distance : Vector3 = direction * projected.length()
+			var projected := old_pos_global.project(normal)
+			var direction := signf(projected.dot(normal))
+			var distance := direction * projected.length()
```
You definitely should not remove the explicit static typing. At a glance do you know the return types of `project()`, `dot()` and `length()`? Does everyone else?

tehKaiN replied:
Well, the problem was caused by using `sign()` function which returns variant and assignment to `direction: Vector3` failed because `sign()` *chose* to use the float-variant. Using `signf()` enforced the float version, `:=` assignment fixed the type mismatch.

> At a glance do you know the return types of project(), dot() and length()? Does everyone else?

When I frequently use those, yes. When fn unnaturally ends with `i` or `f`, you know it's int/float. When I rarely use them, I can ctrl+click them to see the definition in help, which is suboptimal (on-hover prototype would be better but it's not there yet), but works. Using `:=` helps avoiding variant stuff because doing so will result in parse error. But that's my rationale from my code style and I'll happily change it if project maintainer says so.

It's similar discussion about using var/auto in c#/c/c++. Some ppl use it, some not, I guess.

TokisanGames replied:
> Using signf() enforced the float version, := assignment fixed the type mismatch.

Changing it to get it working is fine, so you should type it as float. You shouldn't reduce the maintainer's typing as done here by stripping out all the vec3s.

TokisanGames commented on addons/waterways/plugin.gd:
```diff
@@ -221,27 +221,28 @@ func _forward_3d_gui_input_river(camera: Camera3D, event: InputEvent) -> int:
-					var result = space_state.intersect_ray(ray_from, ray_from + ray_dir * 4096)
+					var space_state := _edited_node.get_world_3d().direct_space_state
+					var ray_params := PhysicsRayQueryParameters3D.create(ray_from, ray_from + ray_dir * 4096)
+					var result = space_state.intersect_ray(ray_params)
```
`result` needs static typing, which should be a dictionary.

I tell my team to never use `:=` unless explicitly being assigned a type in that line, eg `var dir := Vector3(1,2,3)`. Otherwise, always type it. Especially from all engine returns.

tehKaiN commented:
I've changed my commits so that:

- diff is as small as possible
- changed lines mimic code style of surrounding lines

I rather not engage in discussion about which code style is better, because they are fruitless. Explicit as well as implicit typing are present across the codebase, as well as no typing at all, so I can only guess which one is preferred.
